packages/vscode-extension/src/webview/providers/WorkspaceConfigProvider.tsx

```diff
@@ -30,10 +32,12 @@ const WorkspaceConfigContext = createContext<WorkspaceConfigContextType>({
 
 export default function WorkspaceConfigProvider({ children }: PropsWithChildren) {
   const [config, setConfig] = useState<WorkspaceConfigProps>(INITIAL_WORKSPACE_CONFIG);
+  const { project } = useProject();
 
   useEffect(() => {
     function watchConfigChange(e: WorkspaceConfigProps) {
       document.body.setAttribute("data-use-code-theme", `${e.themeType === "vscode"}`);
+      project.dispatchRotate(e.deviceRotation);
```

Comment:
I don't like that we apply the initial config in `DeviceBase`, but watch for updates here. I'd prefer if both happened in the same place.
Personally, I feel the best place would be in `DeviceSession`, so that `DeviceBase` doesn't have to care about `vscode`-specific APIs.